lib/galaxy/tools/parameters/dynamic_options.py

```diff
@@ -127,9 +127,26 @@ def compare_meta_value( file_value, dataset_value ):
             return file_value == dataset_value
         assert self.ref_name in other_values or ( trans is not None and trans.workflow_building_mode), "Required dependency '%s' not found in incoming values" % self.ref_name
         ref = other_values.get( self.ref_name, None )
-        if not isinstance( ref, self.dynamic_option.tool_param.tool.app.model.HistoryDatasetAssociation ) and not isinstance( ref, galaxy.tools.wrappers.DatasetFilenameWrapper ):
+        is_data = isinstance( ref, galaxy.tools.wrappers.DatasetFilenameWrapper )
+        is_data_list = isinstance( ref, galaxy.tools.wrappers.DatasetListWrapper ) or isinstance( ref, list )
+        is_data_or_data_list = is_data or is_data_list
+        if not isinstance( ref, self.dynamic_option.tool_param.tool.app.model.HistoryDatasetAssociation ) and not is_data_or_data_list:
```

Comment:
I [disagree on that](http://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882) - but ``is_data_list`` is used again further down so it should be declared rather than repeatedly rewriting an expression. Once that needs to be in place - for symmetry it makes sense to define ``is_data``. ``is_data_or_data_list`` is less necessary - but I would hope you are not going to nitpick such a small style disagreement.